dcos/nodeutil/util.go

```diff
@@ -53,6 +57,15 @@ type NodeInfo interface {
 	IsLeader() (bool, error)
 	MesosID(context.Context) (string, error)
 	ClusterID() (string, error)
+	TaskCanonicalID(context.Context, string) (*CanonicalID, error)
+}
+
+// CanonicalID is a unique task id.
+type CanonicalID struct {
+	AgentID      string
+	FrameworkID  string
+	ExecutorID   string
+	ContainerIDs []string
```

Comment:
I know that in Mesos terms, there can be multiple container IDs per task. But in DC/OS a 'task' is only ever a single container. I think this struct should reflect that, and `ContainerIDs []string` should be `ContainerID string`

Reply:
pod task have nested task IDs, see https://github.com/dcos/dcos-cli/blob/master/cli/dcoscli/task/main.py#L272-L336

Reply:
That's true, but the nested value is the parent container. The regular ID is still a unique identifier, regardless of hierarchy.

Reply:
correct, but for logs i need the nested values, since the path to stdout/stderr includes the parent IDs